compiler/noirc_evaluator/src/ssa/opt/unrolling.rs

```diff
@@ -680,16 +691,14 @@ impl Loop {
     ) -> Option<BoilerplateStats> {
         let pre_header = self.get_pre_header(function, cfg).ok()?;
         let (lower, upper) = self.get_const_bounds(function, pre_header)?;
-        let lower = lower.try_to_u64()?;
-        let upper = upper.try_to_u64()?;
         let refs = self.find_pre_header_reference_values(function, cfg)?;
 
         let (loads, stores) = self.count_loads_and_stores(function, &refs);
         let increments = self.count_induction_increments(function);
         let all_instructions = self.count_all_instructions(function);
 
         Some(BoilerplateStats {
-            iterations: (upper - lower) as usize,
+            iterations: (upper - lower).max(0) as usize,
```

Comment:
nit: maybe just a brief comment that we do not allow reverse ranges and this will need to be updated in that case

Reply:
Done 👍